osu.Game/IO/MigratableStorage.cs

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using osu.Framework.Platform;
+
+namespace osu.Game.IO
+{
+    /// <summary>
+    /// A <see cref="WrappedStorage"/> that is migratable to different locations.
+    /// </summary>
+    public abstract class MigratableStorage : WrappedStorage
+    {
+        internal static readonly string[] IGNORE_DIRECTORIES = { "cache" };
+
+        internal static readonly string[] IGNORE_FILES =
+        {
+            "framework.ini",
+            "storage.ini"
+        };
```

Comment:
The obvious way to make this customisable by consumers is by making these into protected get-only virtual properties.